### PR TITLE
React new section4

### PR DESCRIPTION
**Pull request by AnjumolSree**
React new section4 #5

### CODE REVIEW

File: anjumol/.gitignore [hunk 1]
```diff
@@ -7,4 +7,7 @@
 /react/s8/node_modules
 /react/s9/project1/node_modules
 /react/s9/project2/node_modules
-/react/course1/node_modules
\ No newline at end of file
+/react/course1/node_modules
```
Comment (Poojanavgurukul) [inline anjumol/.gitignore:10]: avoid adding  git ignore file

File: anjumol/react_new/section_3/package-lock.json [hunk 2]
```diff
@@ -0,0 +1,2140 @@
+{
+  "name": "react-essentials-start",
+  "version": "0.0.0",
+  "lockfileVersion": 2,
+  "requires": true,
+  "packages": {
```
Comment (Poojanavgurukul) [inline anjumol/react_new/section_3/package-lock.json:6]: also, avoid adding package.json file to GitHub

File: anjumol/react_new/section_3/src/App.jsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    );
+  }
+  return (
+    <div>
+      <Header />
+      <main>
+        <h2>Core Concepts</h2>
+        <section id="core-concepts">
+          <ul>
+            {/* <CoreConcept
```
Comment (Poojanavgurukul) [inline anjumol/react_new/section_3/src/App.jsx:33]: removed the unused code

File: anjumol/react_new/section_3/src/components/Header/Header.css [hunk 4]
```diff
@@ -0,0 +1,28 @@
+header {
+    text-align: center;
+    margin: 3rem 0;
+  }
+  
+  header img {
+    height: 5rem;
+    width: 10rem;
+    object-fit: cover;
+  }
```
Comment (Poojanavgurukul) [inline anjumol/react_new/section_3/src/components/Header/Header.css:10]: also explore the internal css

File: anjumol/react_new/section_3/src/components/TabButton.jsx [hunk 5]
```diff
@@ -0,0 +1,10 @@
+export default function TabButton({children,onSelect , isSelected}){
+
+   
+    return(
+            <li>
+                <button className={isSelected ? 'active': undefined} onClick={onSelect}>{children}</button>
```
Comment (Poojanavgurukul) [inline anjumol/react_new/section_3/src/components/TabButton.jsx:6]: work on the formating

File: anjumol/react_new/section_3/src/index.css [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+  -webkit-text-size-adjust: 100%;
+  min-height: 80rem;
+}
+
+
+main {
+  width: 90%;
+  max-width: 50rem;
+  margin: auto;
+}
```
Comment (Poojanavgurukul) [inline anjumol/react_new/section_3/src/index.css:27]: avoid using index.css

File: anjumol/react_new/section_4/react_essential/src/components/Examples.jsx [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+import TabButton from "./TabButton";
+import { useState } from "react";
+import Section from "./Section";
+import Tabs from "./Tabs";
+
+export default function Examples() {
+  const [selectedTopic, setSelectedTopic] = useState();
+
+  function handleSelect(selectedButton) {
+    // selectedButton => 'components', 'jsx', 'props', 'state'
```
Comment (Poojanavgurukul) [inline anjumol/react_new/section_4/react_essential/src/components/Examples.jsx:11]: removed the unused code

File: anjumol/react_new/section_4/react_essential/src/components/TabButton.jsx [hunk 8]
```diff
@@ -0,0 +1,10 @@
+export default function TabButton({ children, isSelected,...props }) {
+  console.log('TABBUTTON COMPONENT EXECUTING');
```
Comment (Poojanavgurukul) [inline anjumol/react_new/section_4/react_essential/src/components/TabButton.jsx:2]: removed the console

File: anjumol/react_new/section_4/react_essential/src/components/Tabs.jsx [hunk 9]
```diff
@@ -0,0 +1,9 @@
+export default function Tabs({ children, buttons ,ButtonsContainer ='menu'}) {
+    //const ButtonsContainer = buttonContainer;
```
Comment (Poojanavgurukul) [inline anjumol/react_new/section_4/react_essential/src/components/Tabs.jsx:2]: remove this line

File: anjumol/react_new/section_4/react_essential/src/index.jsx [hunk 10]
```diff
@@ -0,0 +1,9 @@
+import ReactDOM from "react-dom/client";
+import React from "react";
+
+import App from "./App.jsx";
+import "./index.css";
+
+const entryPoint = document.getElementById("root");
+ReactDOM.createRoot(entryPoint).render(<App />);
+// ReactDOM.createRoot(entryPoint).render(React.createElement(App));
```
Comment (Poojanavgurukul) [inline anjumol/react_new/section_4/react_essential/src/index.jsx:9]: remove the unused code

File: anjumol/react_new/section_4/react_essential/vite.config.js [hunk 11]
```diff
@@ -0,0 +1,7 @@
+import { defineConfig } from "vite";
+import react from "@vitejs/plugin-react";
+
+// https://vitejs.dev/config/
```
Comment (Poojanavgurukul) [inline anjumol/react_new/section_4/react_essential/vite.config.js:4]: removed the commented code

File: anjumol/react_new/section_3/src/components/TabButton.jsx (hunk 5 above)
Comment (AnjumolSree) [reply anjumol/react_new/section_3/src/components/TabButton.jsx:6]: edited

File: anjumol/react_new/section_4/react_essential/src/components/Examples.jsx (hunk 7 above)
Comment (AnjumolSree) [reply anjumol/react_new/section_4/react_essential/src/components/Examples.jsx:11]: done

**Pull request merged**